cardinal/gamestate/map.go

```diff
@@ -0,0 +1,272 @@
+package gamestate
+
+import (
+	"context"
+
+	"github.com/rotisserie/eris"
+)
+
+type MapStorage[K comparable, V any] struct {
+	internalMap map[K]V
+}
+
+func NewMapStorage[K comparable, V any]() *MapStorage[K, V] {
+	return &MapStorage[K, V]{
+		internalMap: make(map[K]V),
+	}
+}
+
+func (m *MapStorage[K, V]) Keys(_ context.Context) ([]K, error) {
+	res := make([]K, 0)
```

Comment:
I think we should initialize `res` with size `len(m.internalMap)`

Reply:
The linter for some reason wants this at 0. Looking at the performance here there is the potential for multiple malloc calls if done this way though. But whatever the linter won't let this go through if I don't do it.

Reply:
it's because the first argument to `make` specifies size. if you append to it, you'll be adding an item at index len(m.internalMap) + 1. 

you need to change this to `res := make([]K, 0, len(m.internalMap)`